### PR TITLE
Add poetry lockfile consistency check to CI environment

### LINKED ISSUE #2904: Check if poetry lockfile is up to date with pyproject.toml in CI environment
/kind feature

**Describe the solution you'd like**
Sometimes, contributors fail to update the lock file after modifying the pyproject.toml, resulting in an obsolete lockfile.. So the tests will run against the old dependencies which may introduce subtle bugs. Poetry merely displays a warning that the lockfile is out of date, which is easily overlooked among the pile of logs in the CI environment. So we should add a test to check if the poetry lock file is up to date with pyproject.toml. This can be acheived by using `poetry lock --check` command.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

### PR DESCRIPTION
**Pull request by sivanantha321**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #2904 

**Type of changes**
Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Feature/Issue validation/testing**:

Please describe the tests that you ran to verify your changes and relevent result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

- Logs

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Checklist**:

- [ ] Have you added unit/e2e tests that prove your fix is effective or that this feature works?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: .github/workflows/e2e-test.yml [hunk 1]
```diff
@@ -40,8 +40,24 @@ jobs:
...
+
+      - uses: actions/setup-python@v4
+        with:
+          python-version: '3.9'
+
+      - name: Install Poetry and version plugin
+        run: ./test/scripts/gh-actions/setup-poetry.sh
+
+      - name: Check poetry lock file consistency
+        run: ./test/scripts/gh-actions/check-poetry-lockfile.sh
```
Comment (yuzisun) [inline .github/workflows/e2e-test.yml:56]: tried out the script seems like kserve package is not consistent with the lock file
```
moving into folder .
skipping folder .
moving into folder ./xgbserver
~/github/kserve/python/xgbserver ~/github/kserve/python
poetry.lock is consistent with pyproject.toml.
~/github/kserve/python
moving into folder ./aiffairness
~/github/kserve/python/aiffairness ~/github/kserve/python
poetry.lock is consistent with pyproject.toml.
~/github/kserve/python
moving into folder ./.pytest_cache
skipping folder ./.pytest_cache
moving into folder ./lgbserver
~/github/kserve/python/lgbserver ~/github/kserve/python
poetry.lock is consistent with pyproject.toml.
~/github/kserve/python
moving into folder ./paddleserver
~/github/kserve/python/paddleserver ~/github/kserve/python
poetry.lock is consistent with pyproject.toml.
~/github/kserve/python
moving into folder ./plugin
skipping folder ./plugin
moving into folder ./kserve
~/github/kserve/python/kserve ~/github/kserve/python
Error: poetry.lock is not consistent with pyproject.toml. Run `poetry lock [--no-update]` to fix it.
```

Comment (yuzisun) [reply .github/workflows/e2e-test.yml:56]: The github action is supposed to fail in this PR?

Comment (sivanantha321) [reply .github/workflows/e2e-test.yml:56]: Hmm... That's weird. For me the script runs successfully.

**Pull request merged**